packages/widgets/src/Pagination/PageSummary.js

```diff
@@ -0,0 +1,70 @@
+import React from 'react'
+import propTypes from '@dhis2/prop-types'
+import { colors, spacers } from '@dhis2/ui-constants'
+
+import translate from '../translate'
+
+const getItemRange = (page, pageSize, total) => {
+    /**
+     * Suppose page is 2 and pageSize is 50 the calculations below
+     * will produce: firstItem 51 and lastItem 100.
+     * But if the total is 0 (no items found) firstItem and
+     * lastItem should both be 0.
+     * And if we are on the last page, the lastItem should not
+     * be derived from the firstItem but should be equal to the total.
+     * Math.min helps fixes both these special cases.
+     */
+    const firstItem = Math.min((page - 1) * pageSize + 1, total)
```

Comment:
This was still confusing me. I assume that `page` starts with `1`, not `0`, otherwise this wouldn't make much sense.
The comments I was referring to was more like:

```js
// `page - 1` -> Make 1-based pages 0-based
// `pageSize + 1` ->
//    pageSize would be last item of prev page,
//    pageSize + 1 first item of current page
const firstItem = Math.min((page - 1) * pageSize + 1, total)
```

I like your explanation as well though, so please keep that one! I mostly care about adding the information that page is 1-based.

Reply:
you could add steps to make it more clear without having to explain multiple steps in one comment

```js
const pageIndex = page - 1 // 1-based -> 0-based
const offset = pageIndex * pageSize
const firstItem = Math.min(offset + 1, total) // Make 1-based
```